lms/djangoapps/instructor_task/tasks_helper/module_state.py

```diff
@@ -409,12 +409,7 @@ def _get_modules_to_update(course_id, usage_keys, student_identifier, filter_fcn
     """
     def get_student():
         """ Fetches student instance if an identifier is provided, else return None """
-        if student_identifier is None:
-            return None
-
-        student_identifier_type = 'email' if '@' in student_identifier else 'username'
-        student_query_params = {student_identifier_type: student_identifier}
-        return User.objects.get(**student_query_params)
+        return None if not student_identifier else get_student_from_identifier(student_identifier)
```

Comment:
Is there any particular reason you would like use `get_student_from_identifier`? It looks like that has been deprecated in favor of `get_user_by_username_or_email` from `student.models`. Could you use that method?